evalml/tests/conftest.py

```diff
@@ -1088,3 +1094,170 @@ def _imbalanced_data_X_y(problem_type, categorical_columns, size):
+        mock_fit = self._patch_method(
+            "fit", side_effect=mock_fit_side_effect, return_value=mock_fit_return_value
+        )
+        mock_score = self._patch_method(
+            "score", side_effect=mock_score_side_effect, return_value=score_return_value
+        )
+
+        # For simplicity, we will always mock predict_proba and _encode_targets even if the problem is not a
+        # classification problem. For regression problems, we'll mock BinaryClassificationPipeline but it doesn't
+        # matter which one we mock since those methods won't be called for regression.
+        pipeline_to_mock = self._pipeline_class
+        if is_regression(self.problem_type):
+            pipeline_to_mock = "evalml.pipelines.BinaryClassificationPipeline"
+
+        mock_encode_targets = self._patch_method(
+            "_encode_targets",
+            side_effect=lambda y: y,
+            return_value=None,
+            pipeline_class_str=pipeline_to_mock,
+        )
+        mock_predict_proba = self._patch_method(
+            "predict_proba",
+            side_effect=None,
+            return_value=predict_proba_return_value,
+            pipeline_class_str=pipeline_to_mock,
+        )
+
+        mock_optimize = patch(
+            "evalml.objectives.BinaryClassificationObjective.optimize_threshold",
+            return_value=optimize_threshold_return_value,
+        )
+
+        mock_tell = patch("evalml.tuners.skopt_tuner.Optimizer.tell")
+
+        # Unfortunately, in order to set the MagicMock instances as class attributes we need to use the
+        # `with ... ` syntax.
+        with mock_fit as fit, mock_score as score, mock_encode_targets as encode, mock_predict_proba as proba, mock_tell as tell, mock_optimize as optimize:
+            # Can think of `yield` as blocking this method until the computation finishes running
+            yield
+            self.mock_fit = fit
+            self.mock_tell = tell
+            self.mock_score = score
+            self.mock_encode_targets = encode
+            self.mock_predict_proba = proba
+            self.mock_optimize_threshold = optimize
+
+    def run_search(
+        self,
+        automl,
+        score_return_value=None,
+        mock_score_side_effect=None,
+        mock_fit_side_effect=None,
+        mock_fit_return_value=None,
+        predict_proba_return_value=None,
+        optimize_threshold_return_value=None,
+    ):
+        """Short-hand for creating test_context and running search within that test_context."""
+        with self.test_context(
+            score_return_value=score_return_value,
+            mock_score_side_effect=mock_score_side_effect,
+            mock_fit_side_effect=mock_fit_side_effect,
+            mock_fit_return_value=mock_fit_return_value,
+            predict_proba_return_value=predict_proba_return_value,
+            optimize_threshold_return_value=optimize_threshold_return_value,
+        ):
+            automl._SLEEP_TIME = 0.0000001
```

Comment:
is this to decrease the sleep time we already have?

Reply:
Yes!